### ISSUE TITLE
5.1 seems to break some models and texture animations, incompatibility with fluidlogged api

### ISSUE THREAD

**Issue by xemnes** [LoliKingdom/LoliASM#92]
unsure why this is happening, a LOT of textures are no longer animating, such as water and animations on blocks from most mods (it might be all of them? not sure).
a lot of models in immersive engineering are broken, steel posts and other machine multiblocks look very weird
https://i.imgur.com/UsK24Gy.png
conveyor belts are showing missing textures
https://i.imgur.com/62ceaRT.png
as well as most machine blocks from thermal series
https://i.imgur.com/t04KcsU.png

also seems to completely break fluidlogging on all blocks with fluidlogged api.

**Comment by Rongmario** [comment]
Try restarting your game and see if most of those fix, just in case, send me your log file.

**Comment by xemnes** [comment]
sadly restarting didnt fix any of the mentioned problems

heres my log
[latest.log](https://github.com/LoliKingdom/LoliASM/files/8641123/latest.log)
:

**Comment by Rongmario** [comment]
Seems like an OptiFine incompatibility, I'll test.

**Comment by Krutoy242** [comment]
I had same no-texture problems, but after i installed 5.2 or after i just reload game in few days, problem gone.

Through, there is still problem with animation. Yes, im running on optifine
![Pfhxs9juSt](https://user-images.githubusercontent.com/936618/167247564-4bfa81f9-a4f4-4d78-86ac-555430246afd.gif)

**Comment by Rongmario** [comment]
Log file would be great as per usual

**Comment by Krutoy242** [comment]
[debug.log](https://github.com/LoliKingdom/LoliASM/files/8644895/debug.log)

**Comment by xemnes** [comment]
ah yes, i just installed 5.2 and i can confirm most things have been fixed
blocks are once again fluidlogged
models from immersive engineering are no longer looking wacky
ducts are visible
no missing or broken textures

except for the animations which are still static

**Comment by HiiroSaya** [comment]
same with 5.2，portal fire and water are all freeze

**Comment by Krutoy242** [comment]
Update:
When i disable this option, issue gone:
`B:onDemandAnimatedTextures=false`

**Comment by xemnes** [comment]
ah hmm, out of curiosity, how much of an improvement on performance does this option give when enabled?

**Comment by Rongmario** [comment]
Tried replicating but animations are all fine with `onDemandAnimatedTextures = true`.

**Comment by xemnes** [comment]
is that with optifine? if not, then i will try narrow down which mod may be causing it

**Comment by Rongmario** [comment]
No, not with Optifine. Optifine has its own `Smart Animations` which does the more or less the same thing but more invasively. You can turn off LoliASM's `onDemandAnimatedTextures` when using Optifine's `Smart Animations`.

**Comment by xemnes** [comment]
ah right, ok. thanks

**Comment by Rongmario** [comment]
Optifine invasively patches minecraft classes. I haven't had time to check if the patches I make conflict with OptiFine's, but it probably is.

**Comment by Krutoy242** [comment]
> Optifine invasively patches minecraft classes

Little bit offtop, but i though you just copy-past VanillaFix code to Loliasm 😅
But seems like its not, because VanillaFix haven't such problem.

**Comment by Rongmario** [comment]
OptiFine overwrite Minecraft's classes, simple-as. And no, if I just copied over VF's code, we would still have the issue such as Tinker's smelteries not having animated textures.

**Comment by xemnes** [comment]
can confirm that was an issue, i had to disable it in vanillafix to fix that issue with smelteries. optifine has always been known to be extremely mod unfriendly, and it doesnt help it is closed source also.

**Comment by Aeron216** [comment]
Can confirm that this setting prevents most blocks from displaying their animations no matter what (vanilla fire, chisel blocks, thermal fluids, techguns items).
And unfortunately disabling this setting isn't an option for users with AMD GPUs, the FPS hit is too high unless they run Optifine's "Smart Animations".

**Comment by Rongmario** [comment]
Closing to make #113 the main `onDemandAnimatedTextures` thread.
